packages/@aws-cdk/aws-ecs/lib/log-drivers/generic-log-driver.ts

```diff
@@ -0,0 +1,60 @@
+import { Construct } from '@aws-cdk/core';
+import { ContainerDefinition } from '../container-definition';
+import { LogDriver, LogDriverConfig } from "../index";
+
+/**
+ * A log driver that sends logs to the specified driver.
+ */
+export class GenericLogDriver extends LogDriver {
+  /**
+   * The log driver to use for the container. The valid values listed for this parameter are log drivers
+   * that the Amazon ECS container agent can communicate with by default. You cannot use awslogs with the GenericLogDriver.
+   * You must use the AwsLogDriver if you want to use awslogs.
+   *
+   * For tasks using the Fargate launch type, the supported log drivers are awslogs and splunk.
+   * For tasks using the EC2 launch type, the supported log drivers are awslogs, syslog, gelf, fluentd, splunk, journald, and json-file.
+   *
+   */
+  private logDriver: string;
+
+  /**
+   * The configuration options to send to the log driver.
+   */
+  private options: { [key: string]: string|number|boolean };
```

Comment:
Can't use union types. We'll have to either type this as `any`, or just fall back to `string`.  I think `string` is probably good enough for the generic log driver.